### PR TITLE
feat: add linter practice for Go

### LINKED ISSUE #415: New Practice: Used Linter for Go lang
## New Practice Proposal

Create a new practice that checks if is used linter for Go lang.

### PR DESCRIPTION
**Pull request by anishkargaonkar**
## Description

Add linter practice for Go project. Checking for the following packages:
- golang.org/x/lint
- github.com/golangci/golangci-lint

## Motivation and Context

resolves #415 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/practices/Go/GoLinterUsedPractice.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import { PracticeEvaluationResult, PracticeImpact, ProgrammingLanguage } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { IPractice } from '../IPractice';
+
+@DxPractice({
+  id: 'Go.LinterUsedPractice',
+  name: 'Use a Go Linter Dependency',
```
Comment (adelkahomolova) [inline src/practices/Go/GoLinterUsedPractice.ts:8]:
```suggestion
  name: 'Use a Go Linter',
```

File: src/practices/Go/GoListerUsedPractice.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  });
+
+  it('Is applicable if it is Go', async () => {
+    containerCtx.practiceContext.projectComponent.language = ProgrammingLanguage.Go;
+
+    const result = await practice.isApplicable(containerCtx.practiceContext);
+    expect(result).toEqual(true);
+  });
+
+  it('Is applicable if it is not Go', async () => {
```
Comment (adelkahomolova) [inline src/practices/Go/GoListerUsedPractice.spec.ts:44]:
```suggestion
  it('Is not applicable if it is not Go', async () => {
```

**Review by adelkahomolova — APPROVED** [review]

**Comment by adelkahomolova** [comment]
@anishkargaonkar Thank you for your pull request! Just accept the suggestions :) 
I have just one question - did you try to run dx-scanner on some repository which is using one of the Go linters? Just to assure, that it's working correctly.

**Comment by anishkargaonkar** [comment]
@adelkahomolova thank you for the review, I've made the suggested changes. Yes, I tested all the scenarios on a local go project.

**Comment by adelkahomolova** [comment]
Thank you @anishkargaonkar and sorry for the delay. I have a totally crazy weekend. I need just one another review for my colleagues. :)

**Review by Jakub-Vacek — APPROVED** [review]

**Pull request merged**